### PR TITLE
Make tests create target branches instead of targeting main/master

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3319

### CODE REVIEW

File: test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlow.cs [hunk 1]
```diff
@@ -104,48 +108,67 @@ public async Task Vmr_BackwardFlowTest()
             TestParameters.GitHubTestOrg,
             sourceDirectory: TestRepository.TestRepo1Name);
 
-        // Clone the VMR
+        TemporaryDirectory testRepoFolder = await CloneRepositoryAsync(TestRepository.TestRepo1Name);
         TemporaryDirectory reposFolder = await CloneRepositoryAsync(TestRepository.VmrTestRepoName);
         var newFilePath = Path.Combine(reposFolder.Directory, "src", TestRepository.TestRepo1Name, TestFileName);
 
-        using (ChangeDirectory(reposFolder.Directory))
+        await CreateTargetBranchAndExecuteTest(targetBranchName, testRepoFolder, async () =>
```
Comment (premun) [inline test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlow.cs:115]: Shouldn't the `targetBranchName` be something dynamic but not `master` basically?

File: test/ProductConstructionService.ScenarioTests/CodeFlowScenarioTestBase.cs [hunk 2]
```diff
@@ -69,31 +54,16 @@ protected async Task CheckBackwardFlowGitHubPullRequest(
...
-            // close the PR
-            await GitHubApi.PullRequest.Update(
-                TestParameters.GitHubTestOrg,
-                targetRepoName,
-                pullRequest.Number,
-                new Octokit.PullRequestUpdate
-                {
-                    State = Octokit.ItemState.Closed
-                });
+            var newFile = files.FirstOrDefault(file => file.FileName == $"{testFile}");
```
Comment (premun) [inline test/ProductConstructionService.ScenarioTests/CodeFlowScenarioTestBase.cs:64]: How does the PR get closed if we removed it here?

Comment (premun) [reply test/ProductConstructionService.ScenarioTests/CodeFlowScenarioTestBase.cs:64]: Or does it get merged instead?

Comment (dkurepa) [reply test/ProductConstructionService.ScenarioTests/CodeFlowScenarioTestBase.cs:64]: we delete the target branch, so the PR gets closed automatically

File: test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlow.cs (hunk 1 above)
Comment (dkurepa) [reply test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_CodeFlow.cs:115]: I didn't push my local branch where this was fixed, thanks

**Review by premun — DISMISSED** [review]

**Review by premun — APPROVED** [review]

**Pull request merged**